lib/goldfinger/client.rb

```diff
@@ -7,12 +7,14 @@ module Goldfinger
   class Client
     include Goldfinger::Utils
 
-    def initialize(uri)
+    def initialize(uri, opts = {})
       @uri = uri
+      @scheme = opts.fetch(:ssl, true) ? "https" : "http"
+      @opts = opts.delete(:ssl) || opts
```

Comment:
Sorry, this will deliver wrong results. This will set `@opts` to the value of `opts[:ssl]`. This area should be:

```ruby
@ssl = opts.delete(:ssl) { true }
@scheme = @ssl ? 'https' : 'http'
@opts = opts
```

This is because `Hash#delete` returns the deleted value while mutating the hash.

Reply:
wow, I did a careless mistake